### PR TITLE
Improve validation of VM configuration

### PR DESCRIPTION
**Pull request by rbradford**
Improve the validation of the VM configuration particularly when that VM is created via the REST API.

### CODE REVIEW

File: vmm/src/config.rs [hunk 1]
```diff
@@ -72,6 +70,25 @@ pub enum Error {
     ParseDevicePathMissing,
     /// Failed to parse vsock parameters
     ParseVsock(OptionParserError),
+    /// Failed to validate configuration
+    Validation(ValidationError),
+}
+
```
Comment (sboeuf) [inline vmm/src/config.rs:70]: Typo in the commit message `The configuration came come`

File: vmm/src/vm.rs [hunk 2]
```diff
@@ -25,7 +25,9 @@ extern crate vm_allocator;
 extern crate vm_memory;
 extern crate vm_virtio;
 
-use crate::config::{DeviceConfig, DiskConfig, HotplugMethod, NetConfig, PmemConfig, VmConfig};
+use crate::config::{
```
Comment (sboeuf) [inline vmm/src/vm.rs:28]: Typo in commit message: `When performing an API boot validate that the configuration.`
Seems like the sentence is not complete :)

File: vmm/src/config.rs [hunk 3]
```diff
@@ -635,10 +632,6 @@ impl DiskConfig {
             warn!("poll_queue parameter currently only has effect when used vhost_user=true");
         }
```
Comment (sboeuf) [inline vmm/src/config.rs:633]: Don't we want to tackle these warnings as part of the `validate()` too?

Comment (rbradford) [reply vmm/src/config.rs:633]: Sadly it's not possible to do that as they have default values that get filled in for the API. In particularly these two are not testing if the values are set but whether they have been specified.

We could move to using `Option<>` types for these fields but I hope that we will get `EVENT_IDX` and WCE control in virtio-block soon enough (or that we default to vhost-user anyway.

Comment (sboeuf) [reply vmm/src/config.rs:641]: Yes that's fine, as long as we have the support coming in virtio, we'll be able to get rid of those warnings.

**Pull request merged**